core/src/main/java/org/apache/iceberg/BaseManifestListFile.java

```diff
@@ -20,22 +20,26 @@
 
 import java.io.Serializable;
 import java.nio.ByteBuffer;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.encryption.EncryptionUtil;
+import org.apache.iceberg.util.ByteBuffers;
 
-public class BaseManifestListFile implements ManifestListFile, Serializable {
+class BaseManifestListFile implements ManifestListFile, Serializable {
   private final String location;
-  private final String mdEncryptionKeyID;
-  private final ByteBuffer encryptedKeyMetadata;
-  private ByteBuffer keyMetadata;
+  private final long snapshotId;
+  private final String keyMetadataKeyID;
+  // stored as a byte array to be Serializable
+  private final byte[] encryptedKeyMetadata;
 
-  public BaseManifestListFile(
+  BaseManifestListFile(
       String location,
-      ByteBuffer keyMetadata,
-      String mdEncryptionKeyID,
+      long snapshotId,
```

Comment:
I added `snapshotId` to this class because I've updated encryption to use it as an AAD.